### PR TITLE
Update find offices filter

### PR DESCRIPTION
**Pull request by EAndrey1091**
[(http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1019)]
Removed **_IncludeDeactivated_** from OfficeFindFilter, added **_IsAscendingSort_** and **_IsActive_**, added the if statements at the  method **_FindAsync_** of the class **_OfficeRepository_**.

### CODE REVIEW

File: src/OfficeService.Models.Dto/Requests/Office/Filters/OfficeFindFilter.cs [hunk 1]
```diff
@@ -5,7 +5,10 @@ namespace LT.DigitalOffice.OfficeService.Models.Dto.Requests.Office.Filters
 {
   public record OfficeFindFilter : BaseFindFilter
   {
-    [FromQuery(Name = "includeDeactivated")]
-    public bool IncludeDeactivated { get; set; } = false;
+    [FromQuery(Name = "ascendingsort")]
```
Comment (ekaterinaryzhenkova) [inline src/OfficeService.Models.Dto/Requests/Office/Filters/OfficeFindFilter.cs]: isAscendingSort

File: src/OfficeService.Models.Dto/Requests/Office/Filters/OfficeFindFilter.cs [hunk 2]
```diff
@@ -5,7 +5,10 @@ namespace LT.DigitalOffice.OfficeService.Models.Dto.Requests.Office.Filters
...
+    public bool? IsAscendingSort { get; set; }
+
+    [FromQuery(Name = "active")]
```
Comment (ekaterinaryzhenkova) [inline src/OfficeService.Models.Dto/Requests/Office/Filters/OfficeFindFilter.cs]: isActive

**Review by a5x55555 — APPROVED** [review]

File: src/OfficeService.Data/OfficeRepository.cs [hunk 3]
```diff
@@ -57,9 +57,14 @@ public async Task<(List<DbOffice>, int totalCount)> FindAsync(OfficeFindFilter f
       IQueryable<DbOffice> dbOffices = _provider.Offices
         .AsQueryable();
 
-      if (!filter.IncludeDeactivated)
+      if (filter.IsAscendingSort.HasValue)
       {
-        dbOffices = dbOffices.Where(x => x.IsActive);
+        dbOffices = filter.IsAscendingSort.Value ? dbOffices.OrderBy(o => o.Name) : dbOffices.OrderByDescending(o => o.Name);
```
Comment (vorobevamv) [inline src/OfficeService.Data/OfficeRepository.cs]: "? dbOffices.OrderBy(o => o.Name) " to line 63, ": dbOffices.OrderByDescending(o => o.Name);" to line 64

File: src/OfficeService.Data/OfficeRepository.cs [hunk 4]
```diff
@@ -57,9 +57,14 @@ public async Task<(List<DbOffice>, int totalCount)> FindAsync(OfficeFindFilter f
...
+      }
+
+      if (filter.IsActive.HasValue)
+      {
+        dbOffices = filter.IsActive.Value ? dbOffices.Where(x => x.IsActive) : dbOffices.Where(x => !x.IsActive);
```
Comment (vorobevamv) [inline src/OfficeService.Data/OfficeRepository.cs]: It can be shorter: "dbOffices= dbOffices.Where(x => x.IsActive == filter.IsActive);"

**Review by vorobevamv — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by ekaterinaryzhenkova — APPROVED** [review]

**Review by AnnPostnikova — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Review by AnnaGakh — APPROVED** [review]

**Pull request merged**